harness/determined/common/experimental/experiment.py

```diff
@@ -149,7 +163,8 @@ def wait(self, interval: float = 5.0) -> ExperimentState:
 
         elapsed_time = 0.0
         while True:
-            exp = bindings.get_GetExperiment(sess, experimentId=self._id).experiment
+            exp = self._get(session=sess)
+            self._hydrate(exp)
```

Comment:
This side effect makes me a little uncomfortable. Especially because the hydration isn't relevant to the `wait`.

What do we want our users to expect re: consistency with master? Should they expect that some method calls sync and some don't? Should they expect to always have to sync themselves?

Reply:
discussed offline. we will automatically refresh when we know the data has changed (e.g. update metadata). we don't want to magically update properties, so we will try to make it intuitive so hopefully it will be obvious to users. for other cases, we should and will provide documentation around the properties lifecycles.

Reply:
How should we surface the side-effects to users? Docstring? Flag? Not at all?

Reply:
docstrings for sure. will add.